src/raw.rs

```diff
@@ -292,6 +304,56 @@ pub fn get_string() -> io::Result<String> {
     }
 }
 
+/// Retrieves a list of file paths from the `CF_HDROP` format.
+///
+/// # Pre-conditions:
+///
+/// * [open()](fn.open.html) has been called.
+pub fn get_file_list() -> io::Result<Vec<PathBuf>> {
+    unsafe {
+        let clipboard_data = GetClipboardData(formats::CF_HDROP);
+        if clipboard_data.is_null() {
+            return Err(utils::get_last_error());
+        }
+
+        let _locked_data = {
+            let locked_ptr = GlobalLock(clipboard_data);
+            if locked_ptr.is_null() {
+                return Err(utils::get_last_error());
+            }
+            LockedData(clipboard_data)
+        };
+
+        let num_files = DragQueryFileW(clipboard_data as HDROP, std::u32::MAX, ptr::null_mut(), 0);
+        let mut file_names = Vec::with_capacity(num_files as usize);
+
+        for file_index in 0..num_files {
+            let required_size_no_null = DragQueryFileW(clipboard_data as HDROP, file_index, ptr::null_mut(), 0);
+            if required_size_no_null == 0 {
+                return Err(io::ErrorKind::Other.into());

```

Comment:
I'm not sure what makes most sense here actually. This error shouldn't normally occur unless there is a programming error, so maybe we should panic here? Same with the other `io::ErrorKind::Other` below.

Reply:
To be honest I'm not sure either since docs don't elaborate on unsuccessful cases

But just to be on safe size let's return `Err(utils::get_last_error())`

Reply:
The problem is, if a function is not documented to set the last error, we'll get the previously set last error instead, which has nothing to do with the current call, so I don't think that's an option. See: https://msdn.microsoft.com/en-us/library/windows/desktop/ms679360(v=vs.85).aspx

Reply:
Well, actually I think it would reset error to 0.
But behavior of `GetLastError` is actually not very specified, so I wouldn't bother too much.

The error `Other` makes no sense because it doesn't really provide anything useful while `GetLastError` will return 0 if everything is fine

Reply:
The `GetLastError` docs are specific about this, though:
> If the function is not documented to set the last-error code, the value returned by this function is simply the most recent last-error code to have been set; some functions set the last-error code to 0 on success and others do not.

So we would get an unrelated error message if we use it. I'd rather have a useless error mesage than a wrong one. :)

Reply:
Oh... Ok, then let's leave it as it is.